subprojects/build-cache-packaging/src/main/java/org/gradle/caching/internal/packaging/impl/TarBuildCacheEntryPacker.java

```diff
@@ -156,10 +153,18 @@ private long packTree(String name, TreeType type, FileSystemSnapshot snapshots,
         return packingVisitor.getPackedEntryCount();
     }
 
-    private static void createTarEntry(String path, long size, int mode, TarArchiveOutputStream tarOutput) throws IOException {
+    private static void createTarEntry(String path, long size, HashCode hash, int mode, TarArchiveOutputStream tarOutput) throws IOException {
```

Comment:
This could take a `@Nullable HashCode` and then we wouldn't need a separate method for no content.